src/entities/utils/getPoolStateWithBalancesV2.ts

```diff
@@ -0,0 +1,114 @@
+import {
+    Address,
+    createPublicClient,
+    formatEther,
+    formatUnits,
+    http,
+} from 'viem';
+import { PoolType } from '@/types';
+import { HumanAmount } from '@/data';
+import { CHAINS, VAULT } from '@/utils';
+import { PoolState, PoolStateWithBalances } from '../types';
+import { getSortedTokens } from './getSortedTokens';
+import { vaultV2Abi } from '@/abi';
+
+type MulticallContract = {
```

Comment:
We've already using a type like this in `src/entities/utils/getPoolStateWithBalancesV3.ts` maybe worth reusing?
Also I see there's `getActualSupply` and `getTotalSupply` helpers in `src/utils/tokens.ts` that do similar things but I don't think are being used. Can you remove those?

Reply:
`MulticallContract` is not really needed - I'll remove it.
`getActualSupply` and `getTotalSupply` are no longer being used because I've adapted in favor of multicalls - I'll remove them 👍